src/Elements/Spatial/Grid2d.cs

```diff
@@ -120,12 +120,16 @@ public Grid2d(IList<Polygon> boundaries, Transform transform = null)
         {
             if (transform == null)
             {
-                transform = new Transform();
-            }
-
-            if (!transform.ZAxis.IsParallelTo(Vector3.ZAxis))
-            {
-                throw new ArgumentException("Currently transforms that are not parallel to the XY Plane are not supported.");
+                //if no transform is supplied, calculate one from the normal. 
+                var planeTransform = boundaries.First().Vertices.ToTransform();
+                if (Math.Abs(planeTransform.ZAxis.Dot(Vector3.ZAxis)).ApproximatelyEquals(1)) // we're xy parallel
+                {
+                    transform = new Transform(planeTransform.Origin);
```

Comment:
This is an example of where a note could help. I think what's happening here is that you're saying, if the transform is _almost_ the default, then force it to be the default. If that's correct, it's not super obvious and a note could help.

Reply:
You could also do this as a ternary which may increase the legibility:
```c#
transform = Math.Abs(planeTransform.ZAxis.Dot(Vector3.ZAxis)).ApproximatelyEquals(1) ?
  new Transform(planeTransform.Origin) :
  planeTransform;

Reply:
Finally, I'm not sure if you want to just force this back to default, as `Vertices.Transform(...)` creates a transform with an X axis along the vector between the first and sector point. Forcing this back to the default sets the X axis to the world X axis.

Reply:
In this case this is the desired behavior. This only occurs when a user has *not* explicitly provided an orientation, only geometry. If the curves are up in space somewhere, we take our chances (there's no obvious implicit alignment); if they are XY-parallel, then we use the default which is a world-parallel grid.